lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -144,7 +144,99 @@ TODO
 
 ## Using exceptions for control-flow
 
-TODO
+#### Problem
+
+This anti-pattern refers to code that forces developers to handle exceptions for control-flow. Exception handling itself does not represent an anti-pattern, but this should not be the only alternative available to developers to handle an error in client code. When developers have no freedom to decide if an error is exceptional or not, this is considered an anti-pattern.
+
+#### Example
+
+An example of this anti-pattern, as shown below, is when a library (for example, `MyModule`) forces its clients to use `try .. rescue` statements to capture and evaluate errors. This library does not allow developers to decide if an error is exceptional or not in their applications.
+
+```elixir
+defmodule MyModule do
+  def janky_function(value) do
+    if is_integer(value) do
+      #...
+      "Result..."
+    else
+      raise RuntimeError, message: "invalid argument. Is not integer!"
+    end
+  end
+end
+```
+
+```elixir
+defmodule Client do
+  # Client forced to use exceptions for control-flow.
+  def foo(arg) do
+    try do
+      value = MyModule.janky_function(arg)
+      "All good! #{value}."
+    rescue
+      e in RuntimeError ->
+        reason = e.message
+        "Uh oh! #{reason}."
+    end
+  end
+end
+```
+
+```elixir
+iex> Client.foo(1)
+"All good! Result...."
+iex> Client.foo("lucas")
+"Uh oh! invalid argument. Is not integer!."
+```
+
+#### Refactoring
+
+Library authors should guarantee that clients are not required to use exceptions for control-flow in their applications. As shown below, this can be done by refactoring the library `MyModule`, providing two versions of the function that forces clients to use exceptions for control-flow: 
```

Comment:
```suggestion
Library authors should guarantee that users are not required to use exceptions for control flow in their applications. As shown below, this can be done by refactoring `MyModule`, providing two versions of the function that forces clients to use exceptions for control flow: 
```